include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_scan.h

```diff
@@ -127,13 +129,24 @@ single_pass_scan_impl(sycl::queue __queue, _InRange&& __in_rng, _OutRange&& __ou
     ::std::size_t num_workitems = num_wgs * wgsize;
 
     constexpr int status_flag_padding = SUBGROUP_SIZE;
-    std::uint32_t status_flags_size = num_wgs + status_flag_padding + 1;
-    std::uint32_t tile_sums_size = num_wgs + status_flag_padding;
+    std::size_t status_flags_elems = num_wgs + status_flag_padding + 1;
+    std::size_t status_flags_size = status_flags_elems * sizeof(std::uint32_t);
+
+    std::size_t tile_sums_elems = num_wgs + status_flag_padding;
+    std::size_t tile_sums_size = status_flags_elems * sizeof(std::uint32_t);
+
+    std::size_t extra_mem_for_aligment = status_flags_size % alignof(_Type);
```

Comment:
Should be the complement of this, no? At present, if `status_flags_size = 5` and `alignof(_Type) == 4`, we have `5 % 4 = 1`, so we pad by one byte to `6` but that's still not aligned.

I think you want:
``` c++
std::size_t extra_mem_for_aligment =  alignof(_Type) - (status_flags_size % alignof(_Type));
```

Reply:
No, now size is the actual size of the memory, and elems the previous "status_flags_size". So in this case, it's (5*4) % 4 = 0. We don't really need padding there I think

Reply:
I would revert the variable names, but we were using size before for number of elements. What do you recommend here?

Reply:
What if `sizeof(_Type) == 16`?

Reply:
I stand corrected :sweat_smile: